pkg/daemon/metrics.go

```diff
@@ -0,0 +1,26 @@
+package daemon
+
+import (
+	"net/http"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var DefaultBindAddress = ":8080"
+
+func StartMetricsListener(addr string, stopCh chan struct{}) {
+	if addr == "" {
+		addr = DefaultBindAddress
+	}
+
+	glog.Infof("Starting metrics listener on %s", addr)
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.Handler())
+	func() {
+		if err := http.ListenAndServe(addr, mux); err != nil {
+			glog.Exitf("Unable to start metrics listener: %v", err)
+		}
+	}()
+	<-stopCh
```

Comment:
You could `select` on stopCh in a go routine and stop the http server when the channel is closed.